ext/bcmath/bcmath.c

```diff
@@ -174,8 +174,8 @@ PHP_FUNCTION(bcadd)
 	bc_init_num(&first);
 	bc_init_num(&second);
 	bc_init_num(&result);
-	php_str2num(&first, ZSTR_VAL(left));
-	php_str2num(&second, ZSTR_VAL(right));
+	php_str2num(&first, ZSTR_VAL(left), 1);
```

Comment:
php_str2num should return whether conversion was successful. On failure, we should do an early `RETURN_THROWS();` here. Otherwise the rest of the function will continue to execute. If nothing else, this will result in a double exception if both arguments are not well-formed.

Reply:
@nikic Do I need to free `first` and `second` before returning?

Reply:
Actually, all three numbers should be `bc_num_free()`d on failure. Might make sense to introduce some `goto finish;` (or similar) here. Still the function must only `RETURN_THROWS()` on failure, so the failure would need to get stored in a local variable for this to work.